### PR TITLE
Update copyrights on all files

### PR DESCRIPTION
**Pull request by pavanbalaji**
## Pull Request Description

Update copyrights on all files

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by pavanbalaji** [comment]
test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by pavanbalaji** [comment]
This PR only has the emacs mode lines for all files.  If someone can tell me the vim mode lines, I can add them too.

**Comment by pavanbalaji** [comment]
test:whitespace

**Comment by pavanbalaji** [comment]
test:jenkins/warnings

**Comment by pavanbalaji** [comment]
test:jenkins/warnings

**Comment by pavanbalaji** [comment]
test:jenkins/warnings

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/most

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/most

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/tcp

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/tcp

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/tcp

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/tcp

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/tcp

**Comment by pavanbalaji** [comment]
@hzhou or @raffenet 

Can you take a look at what's going on here?  Jenkins seems to be misbehaving.

**Comment by hzhou** [comment]
> Can you take a look at what's going on here? Jenkins seems to be misbehaving.

Something changed in github:
```
 > git fetch --tags --progress http://github.com/pmodels/mpich +refs/pull/*:refs/remotes/origin/pr/* # timeout=10
ERROR: Error fetching remote repo 'origin'
```
, which used to work, but stopped working. I have a workaround -- configure the tests to fetch from the author branch directly instead. I'll update the review tests.

**Comment by hzhou** [comment]
test:jenkins/ch3/tcp

**Comment by hzhou** [comment]
I think the review tests are now working.

**Comment by pavanbalaji** [comment]
> I think the review tests are now working.

It still seems to be failing, but with a different error now.

**Comment by pavanbalaji** [comment]
> > I think the review tests are now working.
> 
> It still seems to be failing, but with a different error now.

Nevermind.  It seems to be working.  I was looking at the old tests.

**Comment by pavanbalaji** [comment]
It seems to be working fine now.  Running the remaining tests:

test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by pavanbalaji** [comment]
Forgot to force push my actual changes.  Trying again.

test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by hzhou** [comment]
>  test/mpi/maint/jenkins/set-xfail.sh → test/mpi/maint/jenkins/set-xfail.zsh 

Does this script has any `zsh` specific features? That script is migrated from Jenkins script and I think zsh was chosen due to a random personal preference. If bash works, maybe we should stick to bash rather than introducing yet another dependency.

**Comment by pavanbalaji** [comment]
> > test/mpi/maint/jenkins/set-xfail.sh → test/mpi/maint/jenkins/set-xfail.zsh
> 
> Does this script has any `zsh` specific features? That script is migrated from Jenkins script and I think zsh was chosen due to a random personal preference. If bash works, maybe we should stick to bash rather than introducing yet another dependency.

I didn't add that dependency.  The script was always `zsh`, even though it was named `sh`.  If it's not relying on any `zsh` features, it can be changed to `bash`.  But that's irrelevant to this PR and can be done separately.  This PR corrected the name, just to make it simpler for my script that was setting the copyright headers.

**Comment by hzhou** [comment]
In the spirit of reducing boilerplate clutter, do we need the modeline? I use vim and I never had trouble editing without the modeline. In my opinion, we should work on whitespace checker to guard against tab/space mixing or bad indentations -- if those are issues -- rather than dictate editor settings.

**Comment by hzhou** [comment]
Similar to renaming shell scripts, I personally likes to name all perl scripts with explicit `.pl` extension.

**Comment by pavanbalaji** [comment]
> In the spirit of reducing boilerplate clutter, do we need the modeline? I use vim and I never had trouble editing without the modeline. In my opinion, we should work on whitespace checker to guard against tab/space mixing or bad indentations -- if those are issues -- rather than dictate editor settings.

That's fine, if that's what everyone agrees on.  I can easily add/remove mode lines.  But I want that to be a group decision and not one person's preference.

**Comment by hzhou** [comment]
The test failures probably due to the xfail script name change (to `.zsh`).

**Comment by pavanbalaji** [comment]
> The test failures probably due to the xfail script name change (to `.zsh`).

This is a very silly design that the jenkins testing infrastructure relies on this file, while the mpich source code itself doesn't.  That file should be moved into the jenkins testing repo.

**Comment by hzhou** [comment]
> 
> 
> > The test failures probably due to the xfail script name change (to `.zsh`).
> 
> This is a very silly design that the jenkins testing infrastructure relies on this file, while the mpich source code itself doesn't. That file should be moved into the jenkins testing repo.

I kinda agree. I'll make that change...

**Comment by hzhou** [comment]
test:jenkins/ch4/ucx

**Comment by hzhou** [comment]
@pavanbalaji I moved the `set-xfail` script back to jenkins-script repo, and it looks like it's working. You may delete the script from mpich.

**Comment by wesbland** [comment]
Is this patch backward/forward compatible with the new contributors' agreement or do we all need to sign it first?

**Comment by pavanbalaji** [comment]
> Is this patch backward/forward compatible with the new contributors' agreement or do we all need to sign it first?

This change has nothing to do with the new contributors' agreement.  It simply unifies the copyright to one place, removes developer names and adds institute names in the top-level file instead (I did not see credits for individual developers who were not representing their company anywhere).

**Comment by wesbland** [comment]
> This change has nothing to do with the new contributors' agreement. It simply unifies the copyright to one place, removes developer names and adds institute names in the top-level file instead (I did not see credits for individual developers who were not representing their company anywhere).

👍 Fine with me. I was just making sure.

**Comment by pavanbalaji** [comment]
After some internal discussion, I have removed the mode lines for all files (both emacs and vim).  The intent is that developers will maintain their own formatting requirements within their `.emacs` or `.vimrc` files.

Also removed the `set-xfail.sh` script from the repo.

Rerunning the tests once more:

test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by pavanbalaji** [comment]
Ended up with a conflict with one of the files because of another merge.  Rebased on to `mpich/master`.  Retesting.

test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by pavanbalaji** [comment]
test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by pavanbalaji** [comment]
@hzhou Can you review this PR?  I don't know what "review" means for such changes that are entirely done through scripts.  I guess you can look at the general intent of the commits?

File: COPYRIGHT [hunk 1]
```diff
@@ -6,7 +6,7 @@ which must be included in the prologue of the code and in all source listings
 of the code.
 
 Copyright Notice
- + 2002 University of Chicago
+2020, Argonne National Laboratory
```
Comment (hzhou) [inline COPYRIGHT]: We are expected to update this every year, right?

Some projects use a year range, as some of the code are old and some are new. Should we use `2002 - 2020`?

File: COPYRIGHT [hunk 2]
```diff
@@ -37,3 +37,18 @@ warranty express or implied, or assumes any legal liability or responsibility
...
+ * Intel Corporation
+ * Cray
+ * IBM Corporation
+ * Microsoft Corporation
+ * Mellanox Technologies Ltd.
+ * DataDirect Networks.
+ * Oak Ridge National Laboratory
+ * Sun Microsystems, Lustre group
+ * Dolphin Interconnect Solutions Inc.
+ * Institut Polytechnique de Bordeaux
```
Comment (hzhou) [inline COPYRIGHT]: So much better!

File: doc/mpich/sm2dot [hunk 3]
```diff
@@ -1,7 +1,9 @@
-#!/usr/bin/env perl
-#
-# sm2dot
```
[the remark this answers is not in the document]
Comment (pavanbalaji) [reply doc/mpich/sm2dot]: No, they were an intentional part of my script to clean up useless header content.

File: COPYRIGHT (hunk 1 above)
Comment (pavanbalaji) [reply COPYRIGHT]: We could use a range.  I'll add that.  I need to go back to CVS and find the earliest commit.

File: COPYRIGHT (hunk 2 above)
Comment (wesbland) [inline COPYRIGHT]:
```suggestion
 * Institut Polytechnique de Bordeaux
 
 Contributions copyright their original authors.
```

This is one thing that's missing if we're trying to replicate the per-file headers. I think @pavanbalaji has another version of this text that might be better.

Comment (pavanbalaji) [reply COPYRIGHT]: I'll add it in shortly.

**Comment by pavanbalaji** [comment]
I have marked this PR as do-not-merge.  Whatever changes we make have to be signed off by our legal teams before I push this change in.

**Comment by pavanbalaji** [comment]
test:jenkins/warnings
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by pavanbalaji** [comment]
test:mpich/warnings
test:mpich/ch3/most
test:mpich/ch4/most

**Comment by pavanbalaji** [comment]
test:mpich/warnings
test:mpich/ch3/most
test:mpich/ch4/most

**Comment by hzhou** [comment]
If nothing significant has changed since a month ago, I think I am ready to approve it. I'll just wait for the tests to clear. Meanwhile, @pavanbalaji , if there is any particular area you'd like me to double check, let me know.

**Comment by pavanbalaji** [comment]
> If nothing significant has changed since a month ago, I think I am ready to approve it. I'll just wait for the tests to clear. Meanwhile, @pavanbalaji , if there is any particular area you'd like me to double check, let me know.

Nothing particular to review.  Maybe a quick glance is sufficient.  This is almost entirely scripted.  But the tests are not getting triggered for some reason.

**Pull request merged**